harvester/NexusRepository.py

```diff
@@ -0,0 +1,282 @@
+            record["identifier"] = nexus_dats_json["_self"]
+
+        # Delete deprecated records
+        if "_deprecated" in nexus_record and nexus_record["_deprecated"]:
+            return False
+
+        # Item URL
+        if ("conp_portal_website" in nexus_record) and nexus_record["conp_portal_website"]:
+            if ("@id" in nexus_record["conp_portal_website"]) and nexus_record["conp_portal_website"]["@id"]:
+                record["item_url"] = nexus_record["conp_portal_website"]["@id"]
+
+        if "item_url" not in record:  # use sdo:distribution accessMode (usually OSF URL) as backup item_url
+            if ("sdo:distribution" in nexus_record) and nexus_record["sdo:distribution"]:
+                distribution_urls = []
+                distribution_list = nexus_record["sdo:distribution"]
+                if not isinstance(distribution_list, list):
+                    distribution_list = [distribution_list]
+                for distribution in distribution_list:
+                    if ("sdo:accessMode" in distribution) and distribution["sdo:accessMode"]:
+                        if ("valueIRI" in distribution["sdo:accessMode"]) and distribution["sdo:accessMode"]["valueIRI"]:
+                            distribution_urls.append(distribution["sdo:accessMode"]["valueIRI"])
+                distribution_urls = list(set(distribution_urls))
+                if len(distribution_urls) == 1:
+                    record["item_url"] = distribution_urls[0]
+            if "item_url" not in record:
+                self.logger.error("Record {} missing item_url".format(record["identifier"]))
+                return None
+
+        # Description
+        if ("description" in nexus_record) and nexus_record["description"]:
+            record["description"] = nexus_record["description"]
+
+        # Subjects
+        if ("isAbout" in nexus_record) and nexus_record["isAbout"]:
+            isAbout_list = nexus_record["isAbout"]
+            if not isinstance(isAbout_list, list):
+                isAbout_list = [isAbout_list]
+            record["subjects"] = []
+            for subject in isAbout_list:
+                if ("name" in subject) and subject["name"]:
+                    record["subjects"].append(subject["name"])
+
+        # Rights
+        if ("licenses" in nexus_record) and nexus_record["licenses"]:
+            if ("name" in nexus_record["licenses"]) and nexus_record["licenses"]["name"]:
+                record["rights"] = nexus_record["licenses"]["name"]
+
+        # Title
+        if ("name" in nexus_record) and nexus_record["name"]:
+            record["title"] = nexus_record["name"]
+            if isinstance(record["title"], list):
+                if len(set(record["title"])) == 1:
+                    record["title"] = record["title"][0]
+                else:
+                    self.logger.error("Record {} has multiple titles: {}".format(record["identifier"], record["title"]))
+                    return None
```

Comment:
Should this be a hard error? Not indexing the dataset at all if we found multiple titles?

Reply:
I don't think we actually hit this, but you're right. We could either pick the first one or join them?